src/plugins/kibana/server/lib/schemas/resources/index_pattern_schema.js

```diff
@@ -0,0 +1,23 @@
+const Joi = require('joi');
+
+module.exports = Joi.object({
+  id: Joi.string().required(),
+  title: Joi.string().required(),
+  time_field_name: Joi.string(),
+  interval_name: Joi.string(),
+  not_expandable: Joi.boolean(),
+  fields: Joi.array().items(
```

Comment:
fields needs to have at least one field it would seem https://github.com/hapijs/joi/blob/master/API.md#arrayminlimit